File: net/Azure.Test.PerfStress/ExceptionTest.cs
using System;
using System.Threading;
using System.Threading.Tasks;

namespace Azure.Test.PerfStress
{
    // Measures the overhead of creating, throwing, and catching an exception (compared to NoOpTest)
    public class ExceptionTest : PerfStressTest<PerfStressOptions>
    {
        public ExceptionTest(PerfStressOptions options) : base(options) { }

        public override void Run(CancellationToken cancellationToken)
        {
            try
            {
                throw new InvalidOperationException();
            }
            catch
            {
            }
        }

        public override Task RunAsync(CancellationToken cancellationToken)
        {
            try
            {
                throw new InvalidOperationException();
            }
            catch
            {
            }

            return Task.CompletedTask;
        }
    }
}

File: sdk/core/azure-perfstress/src/main/java/com/azure/perfstress/ExceptionTest.java
package com.azure.perfstress;

import reactor.core.publisher.Mono;

public class ExceptionTest extends PerfStressTest<PerfStressOptions> {
    public ExceptionTest(PerfStressOptions options) {
        super(options);
    }

    @Override
    public void Run() {
        try {
            throw new IllegalArgumentException();
        }
        catch (Exception ex) {
        }
    }

    @Override
    public Mono<Void> RunAsync() {
        try {
            throw new IllegalArgumentException();
        }
        catch (Exception ex) {
        }

        return Mono.empty();
    }
}